cirq/experiments/fidelity_estimation.py

```diff
@@ -470,17 +505,42 @@ def __enter__(
     def __exit__(self, exc_type, exc_val, exc_tb):
         pass
 
-    def update(self, increment: int):
+    def update(self, n: int = 1):
         pass
 
 
+@dataclass(frozen=True)
+class _ZippedCircuit:
+    """A fully-wide circuit made by zipping together a bunch of two-qubit circuits
+    and its provenance data.
+
+    Args:
+        wide_circuit: The zipped circuit on all pairs
+        pairs: The pairs of qubits operated on in the wide circuit.
+        layer_i: The index of the GridInteractionLayer to which the pairs correspond.
```

Comment:
I'm having trouble understanding why this code is so tightly coupled with those interaction layers? The layers are static but in the end this code will need to run with completely dynamic, moment-dependent configurations. What is relation between those two?

Reply:
This might become clear in a following PR, but let me try to explain and then I'll think about how to make it clearer in code.

The primary idea is that you have a number of different moments that define the spacial layout of simultaneous two-qubit gates. The essential essence of each of those moments/layers is just the list of pairs, here as `pairs`. Indeed, `pairs` (and the circuit) is really the only essential part of this dataclass. The other attributes are metadata to maintain the provenance of the particular set of pairs. 

Previous XEB implementations relied heavily on these `GridInteractionLayer` objects to semi-implicitly define the set of pairs by making several assumptions (like you're operating on a nice grid, all the couplings on the grid work, you'll be using the whole device and/or the calling function will downsample only the pairs that actually exist). So in traditional XEB there are always four layers (one for each of the N, S, E, W set of pairs) and they're represented by these objects.

Of course, for the calibration API we re-purpose `cirq.Moment` as a "layer". What you'll see in a following PR is that these fields will either contain a `GridInteractionLayer`, a `Moment`, or even `None` since it's just metadata that doesn't really matter as long as you fill in `pairs`. 

`layer_i` and `combination_i` are just propagated through to be part of the result object. Let me see if I can clarify that

Reply:
I'm changing the docstrings to be more clear about what `layer_i` and `combination_i` are. I've added a note to both that they don't modify any behavior, but are just propagated to the output. I'm also re-ordering the fields in this dataclass to put them last, because they are less important. I've added a note to `combination` that it's important to be able to "unzip" the circuit.